docs/guides/connecting_to_your_data/database/bigquery.md

```diff
@@ -1,4 +1,179 @@
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+Put your connection string in this template:
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py#L24-L38
+```
+Run this code to test your configuration.
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py#L47
+```
+
+</TabItem>
+<TabItem value="python">
+
+Put your connection string in this template:
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_python_example.py#L24-L41
+```
+Run this code to test your configuration.
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_python_example.py#L46
+```
+
+</TabItem>
+</Tabs>
+
+You will see your database tables listed as `Available data_asset_names` in the output of `test_yaml_config()`.
+
+Feel free to adjust your configuration and re-run `test_yaml_config` as needed.
+
+### 6. Save the Datasource configuration to your DataContext
+
+Save the configuration into your `DataContext` by using the `add_datasource()` function.
+
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py#L49
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_python_example.py#L48
+```
+
+</TabItem>
+</Tabs>
+
+
+### 7. Test your new Datasource
+
+Verify your new Datasource by loading data from it into a `Validator` using a `BatchRequest`.
+
+<Tabs
+  defaultValue='runtime_batch_request'
+  values={[
+  {label: 'Using a SQL query', value:'runtime_batch_request'},
+  {label: 'Using a table name', value:'batch_request'},
+  ]}>
+  <TabItem value="runtime_batch_request">
+
+Here is an example of loading data by specifying a SQL query.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py#L52-L58
+```
+
+:::note
+Currently BigQuery does not allow for the creation of temporary tables as the result of a query.  Therefore, Great Expectations will automatically create a permanent table with a random UUID in the `GCP_PROJECT` and `BIGQUERY_DATASET` defined in the connection string.
+Great Expectations also allows for a named table to be used as a temporary table, and the name can be passed in as a `batch_spec_passthrough` parameter. In the following example we are using a table named `ge_temp`.
+:::  
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/bigquery_yaml_example.py#L61-L76
+```
+
```

Comment:
This is the section where both "versions" of the `batch_spec` (with and without the `batch_spec_passthrough`) are described.  Requesting feedback on how this part looks and flows